### ISSUE TITLE
Inconsistent behavior in tests with search

### ISSUE THREAD

**Issue by kiddick** [cannatag/ldap3#409]
Search with real connection:
``` python
        connection.search(
            search_base=connection.user,
            search_filter='(cn=*)',
            attributes=['pwdChangedTime', 'pwdPolicySubentry'],
        )
```
And say entity result doesn't have `pwdPolicySubentry` that parameter.

So attempt to access to that parameter
```python
>>> connection.entries[0].pwdpolicysubentry
pwdPolicySubentry: <no value>
```
And it is ok. Just no value.
Trying same with mocked connection:
```python
...
entry = {
        'cn': 'common_name',
        'displayName': 'common_name',
        'objectClass': ['top', 'person', 'organizationalPerson'],
    }
connection.strategy.add_entry('dn=..', entry)
connection.search(
    search_base=connection.user,
    search_filter='(cn=*)',
    attributes=['pwdChangedTime', 'pwdPolicySubentry'],
)
connection.entries[0].pwdpolicysubentry
  File "/ldap3-2.3-py3.5.egg/ldap3/abstract/entry.py", line 192, in
__getattr__                                                                                                             
    raise LDAPCursorError(error_message)
ldap3.core.exceptions.LDAPCursorError: attribute 'pwdpolicysubentry' not found 
```

**Comment by kiddick** [comment]
It looks like `response = search_result_entry_response_to_dict(entry, self.connection.server.schema, self.connection.server.custom_formatter, self.connection.check_names)` in mockSync should return empty requested attributes if they meet schema requirements.

**Comment by kiddick** [comment]
Bump.
Any ideas? May be you need more details? Thanks

**Comment by cannatag** [comment]
Hi kiddick, sorry for being so late. I’m currently very busy on another project. Hope to release the 2.4 version for the end of the month.

Is this issue actually blocking you? 

Giovanni

**Comment by kiddick** [comment]
Not really.
But it is cool if it will be fixed with 2.4
Thanks for reply!

**Comment by cannatag** [comment]
Hi, can you check the code in dev? The Mock connection didn't honour the return_empty_attributes parameter of the connection object . It should work now.

**Comment by kiddick** [comment]
Cool! Works for me!

**Comment by cannatag** [comment]
Fixed in v2.4
